### PR TITLE
Update to BaseApp 23.06 and add data checker

### PR DESCRIPTION
**Pull request by sparshg**
Seems like upstream has an icon of 62x60 which is not square resulting in build error, should we make a patch?

### CODE REVIEW

**Comment by tchx84** [comment]
@sparshg yes, please fix whatever it's required.

**Review by tchx84 — CHANGES_REQUESTED** [review]
If I add a few values and try to graph it I get this

```
Traceback (most recent call last):
  File "/app/share/sugar/activities/Chart.activity/activity.py", line 704, in _add_chart_cb
    self.update_chart()
  File "/app/share/sugar/activities/Chart.activity/activity.py", line 826, in update_chart
    self._render_chart()
  File "/app/share/sugar/activities/Chart.activity/activity.py", line 766, in _render_chart
    self.current_chart.render()
  File "/app/share/sugar/activities/Chart.activity/chart.py", line 160, in render
    chart.render()
  File "/app/share/sugar/activities/Chart.activity/sugarpycha/chart.py", line 119, in render
    self._renderAxis(cx)
  File "/app/share/sugar/activities/Chart.activity/sugarpycha/chart.py", line 506, in _renderAxis
    self._renderYTick(cx, tick)
  File "/app/share/sugar/activities/Chart.activity/sugarpycha/chart.py", line 415, in _renderYTick
    return self._renderTick(cx, tick,
  File "/app/share/sugar/activities/Chart.activity/sugarpycha/chart.py", line 363, in _renderTick
    if isinstance(tick, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
```
It's also not preserving the values I add, most likely due to the bug above.

@sparshg , like we talked the other week, please test things carefully

**Comment by sparshg** [comment]
Sorry, fixed the issue, working as expected now.

File: chart-port.patch [hunk 1]
```diff
@@ -157,6 +148,25 @@ index fcb57c0..779400c 100644
...
++        validArgs = inspect.getfullargspec(colorSchemeClass.__init__)[0]
+         kwargs = dict([(k, v)
+                       for k, v in list(kwargs.items()) if k in validArgs])
+         self.colorScheme = colorSchemeClass(keys, **kwargs)
+@@ -360,7 +360,7 @@ class Chart(object):
+ 
+     def _renderTick(self, cx, tick, x, y, x2, y2, rotate, text_position):
+         """Aux method for _renderXTick and _renderYTick"""
+-        if isinstance(tick, collections.Callable):
++        if isinstance(tick, collections.abc.Callable):
```
Comment (tchx84) [inline chart-port.patch:182]: I grep'd the code, if this is the fix, we should probably fix it everywhere else.

```
grep "collections.Callable" -r .
./chart/sugarpycha/bar.py:                if isinstance(self.options.yvals.renderer, collections.Callable):
./chart/sugarpycha/chart.py:        if isinstance(tick, collections.Callable):
./chart/sugarpycha/polygonal.py:        if isinstance(tick, collections.Callable):
./chart/sugarpycha/polygonal.py:        if isinstance(tick, collections.Callable):
./chart/sugarpycha/radial.py:        if isinstance(tick, collections.Callable):
./chart/sugarpycha/radial.py:        if isinstance(tick, collections.Callable):

```

**Review by tchx84 — CHANGES_REQUESTED** [review]

**Review by tchx84 — APPROVED** [review]

**Pull request merged**